openproblems/data/tabula_muris_senis.py

```diff
@@ -12,14 +13,34 @@
 METHOD_ALIASES = {"10x 3' v2": "droplet", "Smart-seq2": "facs"}
 
 
+def check_unknown_organs(datasets, organ_list):
+    known_organs = set([t["label"] for d in datasets for t in d["tissue"]])
+    unknown_organs = set(organ_list) - known_organs
+    if unknown_organs:
+        raise ValueError(
+            f"Unknown organs provided in `organ_list': {', '.join(unknown_organs)}. "
+            f"Known organs are {', '.join(known_organs)}"
+        )
+
+
 def matching_dataset(dataset, method_list, organ_list):
-    # we want to skip combined datasets
-    if len(dataset["tissue"]) > 1:
+    assert len(dataset["assay"]) == 1
+    method = dataset["assay"][0]["label"]
+    method = METHOD_ALIASES[method]
+
+    # if organ_list is not empty, we want specific tissues
+    if organ_list and len(dataset["tissue"]) > 1:
         return False
-    if dataset["tissue"][0]["label"] not in organ_list:
+
+    # if organ_list is not empty, check for specific tissue
+    if organ_list and dataset["tissue"][0]["label"] not in organ_list:
```

Comment:
`if organ_list` will fail if `organ_list` is a `np.ndarray`. Can you do `if organ_list is not None`?

Reply:
Or since you convert to an empty list, `if len(organ_list) > 0`